aspnet/fundamentals/localization.rst

```diff
@@ -0,0 +1,337 @@
+		// Add the platform handler to the request pipeline.
+		app.UseIISPlatformHandler();
+
+		var options = new RequestLocalizationOptions();
+		options.RequestCultureProviders.Clear();
+		var cookieRequestCultureProvider = new CookieRequestCultureProvider { CookieName = "TestCookie" };
+		options.RequestCultureProviders.Add(cookieRequestCultureProvider);
+		app.UseRequestLocalization(options);
+
+		app.Run(async (context) =>
+		{
+			await context.Response.WriteAsync(
+	$@"<!doctype html>
+	<html>
+	<head>
+	<script>
+	function createCookie() {{
+		var cookieValue = '{cookieRequestCultureProvider.CookieName}=c=es-ES|uic=es-ES';
+		document.cookie = cookieValue;
+		window.location = window.location.href.split('?')[0];
+	}}
+
+	function clearCookie() {{
+		document.cookie='{cookieRequestCultureProvider.CookieName}=""""';
+		window.location = window.location.href.split('?')[0];
+	}}
+	</script>
+	</head>
+	<body>");
+
+		await context.Response.WriteAsync($"<h1>{SR["Hello"]}</h1>");
+		await context.Response.WriteAsync($"<input type=\"button\" value=\"Create es-ES culture cookie\" onclick='createCookie();' /> ");
+		await context.Response.WriteAsync($"<input type=\"button\" value=\"Clear es-ES culture cookie\" onclick='clearCookie();' /> ");
+		await context.Response.WriteAsync(
+	@"</body>
+	</html>");
+		});
+	}
+
+Using the AcceptLanguageHeaderRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``AcceptLanguageHeaderRequestCultureProvider`` attempts to determine the culture by inspecting the Accept-Language value in the request header. If an Accept-Language value is not present or doesn't contain at least one culture value, ``AcceptLanguageHeaderRequestCultureProvider`` will return null. In addition, since the Accept-Language value can hold multiple cultures, the ``AcceptLanguageHeaderRequestCultureProvider.MaximumAcceptLanguageHeaderValuesToTry`` property, which defaults to 3, defines the maximum number of cultures to try.
+
+As Internet Explorer (IE) honors your operating system language settings, testing the ``AcceptLanguageHeaderRequestCultureProvider`` is easy. The following screen shot shows the request header in the Fiddler tool after setting the Windows language to Spanish.
+
+.. image:: localization/_static/accept-language-header.png
+
+With the Accept-Language request value being set, you would need to simply specify that the ``AcceptLanguageHeaderRequestCultureProvider`` as the provider you want to use.
+
+.. code-block:: c#
+	:emphasize-lines: 6-9
+
+	public void Configure(IApplicationBuilder app, IStringLocalizer<Startup> SR)
+	{
+		// Add the platform handler to the request pipeline.
+		app.UseIISPlatformHandler();
+
+		var options = new RequestLocalizationOptions();
+		options.RequestCultureProviders.Clear();
+		options.RequestCultureProviders.Add(new AcceptLanguageHeaderRequestCultureProvider() );
+		app.UseRequestLocalization(options);
+
+		...
+	}
+
+Implementing a custom IRequestCultureProvider
```

Comment:
I don't think this particular example adds much value. The example implementation isn't something you would ever do. I would just remove this section unless we have a real scenario we want to demonstrate.